### PR TITLE
add support for multiple routes

### PR DESCRIPTION
**Pull request by wis**
initial fix for #280

### CODE REVIEW

File: packages/deployer-cf-workers/src/deploy.ts [hunk 1]
```diff
@@ -134,6 +134,11 @@ export const deployServer: FabServerDeployer<ConfigTypes.CFWorkers> = async (
   if (workers_dev) {
     return await publishOnWorkersDev(api, account_id, script_name)
   } else {
+    if ('routes' in config) {
+      var promises = config['routes'].map(r => publishOnZoneRoute(api, zone_id,
+                                                                  r, script_name))
+      return await Promise.all(promises)
+    }
     return await publishOnZoneRoute(api, zone_id, route, script_name)
```
[the remark this answers is not in the document]
Comment (geelen) [reply packages/deployer-cf-workers/src/deploy.ts:142]: Or just do `for (route in [...config.routes, config.route])`

File: packages/deployer-cf-workers/src/deploy.ts [hunk 2]
```diff
@@ -165,11 +170,24 @@ function checkValidityForZoneRoutes(config: ConfigTypes.CFWorkers) {
...
+        ${missing_config.map((k) => `💛• ${k}💛`).join('\n')}`)
+    }
+  }
+  if ('routes' in config) {
+    let routes = config['routes']
+    if (!Array.isArray(routes)) {
+      throw new InvalidConfigError(`value for \`routes\` key of @fab/deploy-cf-workers config should be an Array`)
+    }
+    if ([...new Set(routes)].length !== routes.length) {
+      throw new InvalidConfigError(`Duplicate item in value for \`routes\` key of @fab/deploy-cf-workers config`)
```
Comment (geelen) [inline packages/deployer-cf-workers/src/deploy.ts:189]: This is a good idea!

**Review by geelen — COMMENTED** [review]
This is looking pretty good! Not sure where your TS errors are coming from, my way of extracting types and defaults from a single object is maybe not worth pursuing further?

Happy for you to get this as far as you can and I'll finish it off though.

**Comment by wis** [comment]
>This is looking pretty good! 

Thank you!
> Not sure where your TS errors are coming from, my way of extracting types and defaults from a single object is maybe not worth pursuing further?

I'm not experienced with Typescript, I found out that TS has a special type `never[]` for an empty array, so I made the default value for `routes` have one empty string item (`['']`). there must have been a reason you started using TS this way, but I don't see it because I have not created or worked enough on this project/code-base.

File: packages/deployer-cf-workers/src/deploy.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wis) [reply packages/deployer-cf-workers/src/deploy.ts:142]: > think this needs to be in an else? I'd suggest that you should never have both route and routes in a config file, we can catch that early and throw an exception, linking to a docs page that explains.

when you have _both_ keys `route` and `routes` and run `wrangler publish` you get this error:
>Error: you must set EITHER workers_dev = true OR provide a zone_id and route/routes.

I think we should also make it explicit either or.

**Comment by wis** [comment]
@geelen I can't figure out why is the E2E test for Next.js failing, it is unclear from this error message `Command failed with exit code 1: yarn build:fab`:
https://github.com/fab-spec/fab/pull/281/checks?check_run_id=1305312488#step:13:612

**Comment by geelen** [comment]
That build failure isn't you, it's failing on master too, and for no reason I can understand...

Will look into it.

**Pull request merged**